pkg/apis/v1alpha1/serviceimport.go

```diff
@@ -107,6 +108,17 @@ type ServicePort struct {
 
 	// The port that will be exposed by this service.
 	Port int32 `json:"port"`
+
+	// Number or name of the port to access on the pods targeted by the service.
+	// Number must be in the range 1 to 65535. Name must be an IANA_SVC_NAME.
+	// If this is a string, it will be looked up as a named port in the
+	// target Pod's container ports. If this is not specified, the value
+	// of the 'port' field is used (an identity map).
+	// This field is ignored for services with clusterIP=None, and should be
+	// omitted or set equal to the 'port' field.
+	// More info: https://kubernetes.io/docs/concepts/services-networking/service/#defining-a-service
+	// +optional
+	TargetPort intstr.IntOrString `json:"targetPort,omitempty"`
```

Comment:
i asked the same question before in the upstream, [slack discussion](https://kubernetes.slack.com/archives/C09R1PJR3/p1655184999186449)

is it invalid any more?

Reply:
Hmm I didn't see this this thread although, our use case is quite simple: we have derived service in Cilium implementation. If we want the traffic to be routed correctly we need the targetPort property on the derived Service object. Currently targetPort would be the only property that would need us to not reconcile derived services only based of the ServiceImport and this feels weird.

Also ServiceImport was supposed to be able to be implemented (or at least implementable) by kube-proxy at the beginning of the project IIUC or some replacement of kube-proxy. So anything that would attempt to natively support ServiceImport traffic load balancing like a kube-proxy would do on regular services can't do this only via the info available on the API servers currently...